src/Sentry/Internal/PartialStream.cs

```diff
@@ -60,7 +60,8 @@ public override async Task<int> ReadAsync(byte[] buffer, int offset, int count,
             _innerStream.Position = innerPosition;
         }
 
-        var read = await _innerStream.ReadAsync(buffer, offset, actualCount, cancellationToken)
+        // See https://github.com/getsentry/sentry-dotnet/issues/3101
```

Comment:
off, some .NET bug? how can we avoid this from happening again?

Reply:
Don't we have treat-warning-as-error on?

Reply:
> Don't we have treat-warning-as-error on?

We do but the analyzers for this error only appear to trigger in the .NET 10 SDK (we weren't getting any errors before that).